test/utils/format/formatSmart.spec.ts

```diff
@@ -0,0 +1,184 @@
+  test('0', async () => {
+    expect(formatSmart(ZERO, DEFAULT_PRECISION)).toEqual('0')
+  })
+
+  test('1 wei', async () => {
+    expect(formatSmart(ONE, DEFAULT_PRECISION)).toEqual('< 0.001')
+  })
+
+  test('1000 wei', async () => {
+    expect(formatSmart(new BN('1000'), DEFAULT_PRECISION)).toEqual('< 0.001')
+  })
+
+  test('1.23451 Ether, round down', async () => {
+    expect(formatSmart(new BN(toWei(new BN('1234510'), 'microether')), DEFAULT_PRECISION)).toEqual('1.2345')
+  })
+
+  test('1.23456 Ether, round up', async () => {
+    expect(formatSmart(new BN(toWei(new BN('1234560'), 'microether')), DEFAULT_PRECISION)).toEqual('1.2346')
+  })
+
+  test('1.234567890123456789 Ether', async () => {
+    expect(formatSmart(new BN('1234567890123456789'), DEFAULT_PRECISION)).toEqual('1.2346')
+  })
+})
+
+describe('Tokens with precision 6', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1000000')
+    expect(formatSmart(amount, 6)).toEqual('1')
+  })
+
+  test('12,345 units', async () => {
+    const amount = new BN('12345000000')
+    expect(formatSmart(amount, 6)).toEqual('12,345')
+  })
+
+  test('4,567,890.123444 units, use precision 2', async () => {
+    const amount = new BN('4567890123444')
+    expect(formatSmart(amount, 6)).toEqual('4,567,890.12')
+  })
+})
+
+describe('Tokens with precision 0', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1')
+    expect(formatSmart(amount, 0)).toEqual('1')
+  })
+
+  test('12,345 units', async () => {
+    const amount = new BN('12345')
+    expect(formatSmart(amount, 0)).toEqual('12,345')
+  })
+
+  test('4,567,890 units', async () => {
+    const amount = new BN('4567890')
+    expect(formatSmart(amount, 0)).toEqual('4,567,890')
+  })
+})
+
+describe('Tokens with precision 2', () => {
+  test('1 unit', async () => {
+    const amount = new BN('100')
+    expect(formatSmart(amount, 2)).toEqual('1')
+  })
+
+  test('12.34 units', async () => {
+    const amount = new BN('1234')
+    expect(formatSmart(amount, 2)).toEqual('12.34')
+  })
+
+  test('1,234,567.89 units', async () => {
+    const amount = new BN('123456789')
+    expect(formatSmart(amount, 2)).toEqual('1,234,567.89')
+  })
+
+  test('1,234,567.8 units', async () => {
+    const amount = new BN('123456780')
+    expect(formatSmart(amount, 2)).toEqual('1,234,567.8')
+  })
+})
+
+describe('2 decimals', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1000000')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('1')
+  })
+
+  test('12.34 units', async () => {
+    const amount = new BN('12340000')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('12.34')
+  })
+
+  test('4,567,890.123333 units, round down', async () => {
+    const amount = new BN('4567890123333')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('4,567,890.12')
+  })
+
+  test('4,567,890.125555 units, round up', async () => {
```

Comment:
The test is good. But probably you would want to change the test description to --> `4,567,890.125555 units, it doesn't round up`

Reply:
it does round up, from `4,567,890.12555` to `4,567,890.13`

Reply:
values under `10m` keep 2 decimals and round, according to the Gnosis Safe spec